runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -587,6 +588,13 @@ public void start() {
         options.getWindmillHarnessUpdateReportingPeriod().getMillis(),
         TimeUnit.MILLISECONDS);
 
+    workerMessageReportTimer = executorSupplier.apply("WorkerMessageTimer");
+    workerMessageReportTimer.scheduleWithFixedDelay(
+        this::reportPeriodicWorkerMessage,
+        0,
+        options.getWindmillHarnessUpdateReportingPeriod().getMillis(),
```

Comment:
What option would I check against? This also seems more similar to the periodicWorkerUpdates which has no check

Reply:
only schedule here if options.getWindmillHarnessUpdateReportingPeriod() > 0

don't schedule with fixed delay.  Not sure if it would throw exception or just run in tight loop but neither seems particularly desirable and having a way to disable if desired for some reason could be nice.

Reply:
A little confused on the part about not scheduling with fixed delay, the other scheduling option is with fixed period since we need the metrics to be sent every so often like the counter metrics being sent.

Reply:
updated to add check before scheduling, but still using schedule with fixed delay, which seems to be working without any problems